robotframework-ls/src/robotframework_ls/impl/semantic_tokens.py

```diff
@@ -179,16 +282,22 @@ def _tokenize_token(node, initial_token):
             initial_token_type = KEYWORD
 
     if initial_token_type == KEYWORD:
-        token_keyword = _DummyToken(initial_token)
+        token_keyword = _DummyToken(
+            initial_token, RF_TOKEN_TYPE_TO_TOKEN_TYPE_INDEX[KEYWORD]
+        )
         token_gherkin_prefix = extract_gherkin_token_from_keyword(token_keyword)
         if token_gherkin_prefix:
-            yield token_gherkin_prefix, TOKEN_TYPE_TO_INDEX[token_gherkin_prefix.type]
+            yield token_gherkin_prefix, token_gherkin_prefix.index
         token_library_prefix = extract_library_token_from_keyword(token_keyword)
         if token_library_prefix:
-            yield token_library_prefix, TOKEN_TYPE_TO_INDEX[token_library_prefix.type]
-        if token_gherkin_prefix or token_library_prefix:
-            yield token_keyword, RF_TOKEN_TYPE_TO_TOKEN_TYPE_INDEX[KEYWORD]
-            return
+            yield token_library_prefix, token_library_prefix.index
+        is_bdd_statement = keyword_contains_embedded_variable(token_keyword)
```

Comment:
There's no need to check if it contains, just using `ast_utils.tokenize_variables(token)` as explained in other comments should already extract things appropriately without the check.